src/contrib/cluster/Akka.Cluster.Sharding/DDataShard.cs

```diff
@@ -127,7 +127,37 @@ internal sealed class DDataShard : ActorBase, IShard, IWithUnboundedStash, IWith
             }
         }
 
-        public void EntityTerminated(IActorRef tref) => this.BaseEntityTerminated(tref);
+        public void EntityTerminated(IActorRef tref) 
+        {
+            if (!IdByRef.TryGetValue(tref, out var id)) return;
```

Comment:
In the original implementation, `DDataShard` inherits from `RememberingShard` class, not the base `Shard` class through a mixin.
In our implementation, since C# doesn't support mixins, it's implemented with an override. The bug here is that `DDataShard` didn't override the default `EntityTerminated` implementation.